### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by IvanGuculyak**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,38 @@
...
+from app.players.dwarves.dwarf import Dwarf
+from app.players.dwarves.dwarf_blacksmith import DwarfBlacksmith
+from app.players.dwarves.dwarf_warrior import DwarfWarrior
+from app.players.elves.druid import Druid
+from app.players.elves.elf import Elf
+from app.players.elves.elf_ranger import ElfRanger
+
+
+elves_list = []
+dwarves_list = []
```
Comment (Abnormaltype) [inline app/main.py:10]: unnecessary here

File: app/main.py [hunk 2]
```diff
@@ -1 +1,38 @@
...
+        if isinstance(player, DwarfWarrior):
+            dwarves_list.append(player)
+            total_rating += DwarfWarrior.get_rating(player)
+        if isinstance(player, Druid):
+            elves_list.append(player)
+            total_rating += Druid.get_rating(player)
+        if isinstance(player, ElfRanger):
+            elves_list.append(player)
+            total_rating += ElfRanger.get_rating(player)
+    return total_rating
```
Comment (Abnormaltype) [inline app/main.py:28]: you have the same method in all classes. use it.

**Review by Abnormaltype — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 3]
```diff
@@ -1 +1,16 @@
-# write your code here
+from app.players.dwarves.dwarf import Dwarf
+from app.players.elves.elf import Elf
+
+
+def calculate_team_total_rating(players: list) -> int:
```
Comment (katevasilenko) [inline app/main.py:5]:
```suggestion
def calculate_team_total_rating(players: list[Player]) -> int:
```

File: app/main.py [hunk 4]
```diff
@@ -1 +1,16 @@
...
+    return sum(player.get_rating() for player in players)
+
+
+def elves_concert(elves: list) -> None:
+    for elf in elves:
+        Elf.play_elf_song(elf)
```
Comment (katevasilenko) [inline app/main.py:11]:
```suggestion
def elves_concert(elves: list[Elf]) -> None:
    for elf in elves:
        elf.play_elf_song()
```

File: app/main.py [hunk 5]
```diff
@@ -1 +1,16 @@
...
+
+
+def feast_of_the_dwarves(dwarves: list) -> None:
+    for dwarf in dwarves:
+        Dwarf.eat_favourite_dish(dwarf)
```
Comment (katevasilenko) [inline app/main.py:16]: Do the same way as I suggested you in previous function

File: app/players/elves/elf_ranger.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+            musical_instrument: str,
+            bow_level: int
+    ) -> None:
+
+        super().__init__(nickname, musical_instrument)
+        self.__bow_level = bow_level
+
+    def player_info(self) -> str:
+        return f"Elf ranger {self.nickname}. " \
+               f"{self.nickname} has bow of the " \
```
Comment (katevasilenko) [inline app/players/elves/elf_ranger.py:17]: Please avoid to use `\ ` if it possible all over the project

**Review by katevasilenko — CHANGES_REQUESTED** [review]

File: app/players/player.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+class Player(ABC):
+    def __init__(self, nickname: str) -> None:
+        self.nickname = nickname
+
+    @abstractmethod
+    def get_rating(self) -> None:
+        pass
+
+    @abstractmethod
+    def player_info(self) -> None:
```
Comment (olena-hul) [inline app/players/player.py:13]:
```suggestion
    def player_info(self) -> str:

```

**Review by olena-hul — CHANGES_REQUESTED** [review]

**Review by katevasilenko — APPROVED** [review]
Looks good to me now

**Pull request closed without merging**